### DISCUSSION TITLE
Copy gets slower as it executes.

### DISCUSSION THREAD

**Discussion by CamFromStar** [psycopg/psycopg#892] [Q&A]
I am copying a 1.5GB csv file into postgres.  I am using the below code:
```python
with cursor.copy(f"COPY table_a FROM STDIN (FORMAT csv)") as copy:
    while data := file.read(BLOCK_SIZE):
        copy.write(data)
```

I am copying into a temporary table, with no primary key, indexes, triggers, etc.  The process seems to get considerably slower as it goes.  It does the first 250MB in 24 seconds, To get to 1.3GB it has taken 20 minutes.

**Comment by dvarrazzo** [comment]
It likely gets slowed down by the server, not by the client. There is nothing for psycopg to do here.
